### PR TITLE
Bind GCE/AWS credentials and run AWS e2e inside docker

### PR DESCRIPTION
**Pull request by ixdy**
Trying #19 again. 

kubernetes/kubernetes/pull/25376 fixed the issues we encountered the first time.

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -587,27 +588,23 @@
     trigger-job: ''
     timeout: 240
     jenkins_node: 'master'
```
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Does it still need to run on master?

Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: good catch. that's one of the main reasons to do this change in the first place. :)

**Comment by ixdy** [comment]
Seems to work: https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-dockerized-test/73

**Comment by ixdy** [comment]
@fejta @spxtr should we try to merge this sometime today?

**Comment by spxtr** [comment]
I'm fine with it, but @fejta is on call.

File: jenkins/job-configs/kubernetes-jenkins/credentials.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+- wrapper:
```
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/credentials.yaml]: Might appreciate a comment explaining what these are.

Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins/credentials.yaml]: I thought about that, but I wasn't sure that a comment would be any more descriptive than the env var name itself.

Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/credentials.yaml]: Ok, seems reasonable.

**Comment by fejta** [comment]
Before merging I want one of the following to be true:

- This change passes on a staging jenkins instance
- http://submit-queue.k8s.io/#/e2e has had 80% health for the past 24 hours
- The mergebot merged at least 30 PRs in the past 24 hours
- There are no 1.3 PRs in the queue

**Comment by spxtr** [comment]
> This change passes on a staging jenkins instance

It passed in a staging Jenkins job for GCE. I don't think that's enough confidence though, since it hits everything.

**Comment by ixdy** [comment]
I understood what caused the breakage before, and it's been fixed, so I am reasonably confident this is a no-op for most builds. I'm not 100% sure though.

Also, it's not possible to merge 30 PRs in one day. The unit test build takes around 1 hour right now.

**Comment by ixdy** [comment]
given that today is the last day for 1.3 features, it seems reasonable to hold off.

**Comment by ixdy** [comment]
alright, here goes...

**Comment by ixdy** [comment]
well, this broke AWS - we secretly depended on the `aws` binary installed on the Jenkins master:
```
18:39:43 2016/06/28 18:39:43 e2e.go:212: Running: up
18:39:43 ... Starting cluster in us-west-2a using provider aws
18:39:43 ... calling verify-prereqs
18:39:43 Can't find aws in PATH, please fix and retry.
18:39:43 2016/06/28 18:39:43 e2e.go:218: Error running up: exit status 1
```

I'm not sure if we should add this to the `kubekins-test` image or if we should install it on each invocation (like we do with `gcloud`).

**Comment by spxtr** [comment]
Both seem reasonable to me. Is it worth using a new image for these?

Similarly, is it worth using a different image for e2e testing and unit/integration testing?

**Pull request merged**